src/extensions/registries/page-registry.ts

```diff
@@ -0,0 +1,40 @@
+// Extensions-api -> Dynamic pages
+
+import type React from "react";
+import type { RouteProps } from "react-router";
+import type { IconProps } from "../../renderer/components/icon";
+import type { IClassName } from "../../renderer/utils";
+import type { TabRoute } from "../../renderer/components/layout/tab-layout";
+import { BaseRegistry } from "./base-registry";
+import { computed } from "mobx";
+
+export enum PageRegistryType {
```

Comment:
It feels that we should have separate APIs for each type (just to be more futureproof). Ofc that should be done in a separate PR.